contrib/epee/src/connection_basic.cpp

```diff
@@ -128,7 +128,7 @@ connection_basic_pimpl::connection_basic_pimpl(const std::string &name) : m_thro
 int connection_basic_pimpl::m_default_tos;
 
 // methods:
-connection_basic::connection_basic(boost::asio::ip::tcp::socket&& sock, boost::shared_ptr<connection_basic_shared_state> state, ssl_support_t ssl_support)
+connection_basic::connection_basic(boost::asio::ip::tcp::socket&& sock, std::shared_ptr<connection_basic_shared_state> state, ssl_support_t ssl_support)
```

Comment:
My apologies for failing to respond to this earlier, I will revert to `boost::shared_ptr` if desired.

Initially I wrote the white noise code with `std::shared_ptr`, then near the end realized I had to convert existing `boost::shared_ptr` usage to make it work. I switched to `std::shared_ptr` simply because there isn't a need to use boost if the compiler vendors are providing the same functionality. There might be limited cases where `std::shared_ptr` is faster due to some platform specific implementation, but I doubt there is much difference in practice. 

`std::shared_ptr` also has better `noexcept` specifications right now. Boost does appear to follow C++11 specification, but the documentation will simply state "never throws" in several areas instead of stating `noexcept`.

OTOH, using Boost means a single implementation across platforms. Currently we are using `std::shared_ptr` from libstdc++ (Linux, Windows) and libcxx (OSX).